src/3d/symbols/qgspoint3dsymbol_p.cpp

```diff
@@ -85,44 +85,124 @@ class QgsInstancedPoint3DSymbolHandler : public QgsFeature3DHandler
 
   private:
 
-    static Qt3DRender::QMaterial *material( const QgsPoint3DSymbol *symbol, const QgsMaterialContext &materialContext );
-    static Qt3DRender::QGeometryRenderer *renderer( const QgsPoint3DSymbol *symbol, const QVector<QVector3D> &positions );
-    static Qt3DQGeometry *symbolGeometry( const QgsPoint3DSymbol *symbol );
-
     //! temporary data we will pass to the tessellator
-    struct PointData
+    class PointData
     {
-      QVector<QVector3D> positions;  // contains triplets of float x,y,z for each point
+      public:
+
+        PointData() = default;
+
+        float length = 0.0f;
+        float radius = 0.0f;
+        float bottomRadius = 0.0f;
+        float minorRadius = 0.0f;
+        float topRadius = 0.0f;
+        float size = 0.0f;
+        QVector<QVector3D> positions;
+
+        bool operator==( const PointData &other ) const
+        {
+          return ( length == other.length && radius == other.radius
+                   && bottomRadius == other.bottomRadius && minorRadius == other.minorRadius
+                   && topRadius == other.topRadius && size == other.size );
+        }
     };
 
-    void makeEntity( Qt3DCore::QEntity *parent, const Qgs3DRenderContext &context, PointData &out, bool selected );
+    static Qt3DRender::QMaterial *material( const QgsPoint3DSymbol *symbol, const QgsMaterialContext &materialContext );
+    static Qt3DRender::QGeometryRenderer *renderer( const QgsPoint3DSymbol *symbol, const PointData &pointData );
+    static Qt3DQGeometry *symbolGeometry( const QgsPoint3DSymbol *symbol, const PointData &pointData );
+
+    void makeEntity( Qt3DCore::QEntity *parent, const Qgs3DRenderContext &context, QVector<PointData> &out, bool selected );
```

Comment:
Can you add comment on these static functions?

Reply:
Done.